chainer/testing/array.py

```diff
@@ -22,6 +23,25 @@ def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
     try:
         numpy.testing.assert_allclose(
             x, y, atol=atol, rtol=rtol, verbose=verbose)
-    except Exception:
-        print('error:', numpy.abs(x - y).max())
+    except AssertionError:
+        err = numpy.abs(x - y)
+        i = numpy.unravel_index(numpy.argmax(err), err.shape)
+        f = sys.stderr
+        f.write(
+            'assert_allclose failed: \n' +
+            '  shape: {}\n'.format(x.shape) +
+            '  dtype: {} {}\n'.format(x.dtype, y.dtype) +
+            '  i: {}\n'.format(i) +
+            '  x[i]: {}\n'.format(x[i]) +
+            '  y[i]: {}\n'.format(y[i]) +
+            '  err[i]: {}\n'.format(err[i]))
+        f.flush()
+        opts = numpy.get_printoptions()
+        try:
+            numpy.set_printoptions(threshold=numpy.inf)
```

Comment:
Sometimes input array size is very huge. The threshold default value is 1000. 
How about using more reasonable value (For example 10k-1M)?